### ISSUE TITLE
Fix accessibility of the shell's SideBar component

### ISSUE THREAD

**Issue by vanch3d** [hivemq/hivemq-edge#1347]
The sidebar component in the UI Library is failing accessibility (a11y) testing. The main reason is that the implementation of the items in the "menu" has used plain `div` rather than a proper interactive element, breaking the expected patterns of focus and keyboard navigation.

Since the component is used across every HiveMQ frontend product, this issue has a significant impact

## Expected behaviour

- [ ] The component MUST use `Link` or `Button`
- [ ] The component COULD use `List` and `ListItem` for better normalisation
